apps/ai-content-generator/src/components/config/content-type-selection/ContentTypeSelection.tsx

```diff
@@ -0,0 +1,74 @@
+import { Dispatch, useState } from 'react';
+import { Checkbox, FormControl, Table } from '@contentful/f36-components';
+import { ContentTypeProps } from 'contentful-management';
+import { ContentTypeText } from '@components/config/configText';
+import { ContentTypeAction, ContentTypeReducer } from '@components/config/contentTypeReducer';
+import { styles } from './ContentTypeSelection.styles';
+
+interface Props {
+  allContentTypes: ContentTypeProps[];
+  selectedContentTypes: string[];
+  dispatch: Dispatch<ContentTypeReducer>;
+}
+
+const ContentTypeSelection = (props: Props) => {
+  const { allContentTypes, selectedContentTypes, dispatch } = props;
+  const [isSelectAllChecked, setIsSelectAllChecked] = useState(false);
+
+  const allContentTypeIds = allContentTypes.map((ct) => ct.sys.id);
+
+  const handleSelectAll = () => {
+    if (isSelectAllChecked) {
+      dispatch({ type: ContentTypeAction.REMOVE_ALL });
+    } else {
+      dispatch({ type: ContentTypeAction.ADD_ALL, value: allContentTypeIds });
+    }
+
+    setIsSelectAllChecked(!isSelectAllChecked);
+  };
+
+  const handleCheckboxChange = (contentType: ContentTypeProps, isChecked: boolean) => {
+    const action = isChecked ? ContentTypeAction.REMOVE : ContentTypeAction.ADD;
+    dispatch({ type: action, value: contentType.sys.id });
+  };
+
+  const isChecked = (contentTypeId: string) => {
+    return selectedContentTypes.includes(contentTypeId);
```

Comment:
We may want to consider storing the selected content types in a `Set` vs an array. This is creating an O(n^2) on every render.

This would be a pretty easy rewrite, all you'd need to do is to use the `.has()` method here! 

Anywhere you'd actually need the array (like for saving) you would need to just update it to do `Array.from(selectedContentTypes)` to access it as an array. 

And in places where you're adding/deleting there are methods on Set `.delete()` and `.add()`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set